controllers/actions.github.com/autoscalinglistener_controller.go

```diff
@@ -345,7 +346,25 @@ func (r *AutoscalingListenerReconciler) createServiceAccountForListener(ctx cont
 }
 
 func (r *AutoscalingListenerReconciler) createListenerPod(ctx context.Context, autoscalingRunnerSet *v1alpha1.AutoscalingRunnerSet, autoscalingListener *v1alpha1.AutoscalingListener, serviceAccount *corev1.ServiceAccount, secret *corev1.Secret, logger logr.Logger) (ctrl.Result, error) {
-	newPod := r.resourceBuilder.newScaleSetListenerPod(autoscalingListener, serviceAccount, secret)
+	var envs []corev1.EnvVar
+	if autoscalingListener.Spec.Proxy != nil {
+		proxyEnv, err := autoscalingListener.Spec.Proxy.EnvVars(func(s string) (*corev1.Secret, error) {
+			var secret corev1.Secret
+			err := r.Get(ctx, types.NamespacedName{Name: s, Namespace: autoscalingRunnerSet.Namespace}, &secret)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get secret %s: %w", s, err)
+			}
+			return &secret, nil
+		})
+		if err != nil {
+			logger.Error(err, "Unable to get proxy env vars")
+			return reconcile.Result{}, fmt.Errorf("failed to get proxy env vars: %w", err)
+		}
+
+		envs = append(envs, proxyEnv...)
```

Comment:
do we need to use secret for the env, since it may contain a username/password about the proxy auth?

Reply:
Eventually we do yes, to build the URL and set it as an env var (or read the authenticated URL directly if we change the spec). It's pretty standard to surface secrets as env vars in pods though, the only way around this would be putting the secret in a file and mount that inside a volume.

Reply:
do we need to put the `http(s)_proxy` env with auth into a secret we create and reference it via the following?
```
    - name: https_proxy
      valueFrom:
        secretKeyRef:
          name: proxyAuthUrl
          key: https_proxy
```

Otherwise, I think anyone with `Get` pod permission can get the proxy auth by reading the env, will that be a problem?

Reply:
Ha I see what you mean now, yeah that sounds about right 🤦 . Thanks for pointing out!

Reply:
To piggyback on that, if the key to CredentialSecretRef, or even its value, changes, we'll have to be able to propagate that change all the way down to the runner pods.